site_scons/prereq_tools/base.py

```diff
@@ -340,9 +340,12 @@ def checkout_commit(self, subdir):
     def apply_patches(self, subdir, patches):
         """ git-apply a certain hash """
         if patches is not None:
-            for patch in patches:
+            for patch in patches.keys():
                 print("Applying patch %s" % (patch))
-                commands = ['git apply %s' % (patch)]
+                subdir_option = ""
+                if patches[patch] is not None:
+                    subdir_option = "--directory %s" % patches[patch]
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string

Reply:
We've disabled this pylint warning in the other branches.  It can be ignored.